corehq/apps/app_manager/views/utils.py

```diff
@@ -172,16 +172,15 @@ def overwrite_app(app, master_build, report_map=None):
             app_json[key] = value
     app_json['version'] = master_json['version']
     wrapped_app = wrap_app(app_json)
-    for module in wrapped_app.modules:
-        if isinstance(module, ReportModule):
-            if report_map is not None:
-                for config in module.report_configs:
-                    try:
-                        config.report_id = report_map[config.report_id]
-                    except KeyError:
-                        raise AppEditingError(config.report_id)
-            else:
-                raise AppEditingError('Report map not passed to overwrite_app')
+    for module in wrapped_app.get_report_modules():
+        if report_map is not None:
```

Comment:
nit: you could probably have pulled this `if` statement out of this loop so you don't check this for every module (and you save one level of indentation)